core/src/test/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolverTests.java

```diff
@@ -832,7 +831,7 @@ public void testIsPatternMatchingAllIndicesNonMatchingSingleExclusion() throws E
     }
 
     public void testIsPatternMatchingAllIndicesMatchingTrailingWildcardAndExclusion() throws Exception {
-        String[] indicesOrAliases = new String[]{"index*", "-index1", "+index1"};
+        String[] indicesOrAliases = new String[]{"index*", "-index1", "index1"};
         String[] concreteIndices = new String[]{"index1", "index2", "index3"};
         MetaData metaData = metaDataBuilder(concreteIndices);
         assertThat(indexNameExpressionResolver.isPatternMatchingAllIndices(metaData, indicesOrAliases, concreteIndices), equalTo(true));
```

Comment:
I think the tests that you deleted around '+' should be added back as a separate method. You did that for wildcard expressions but not for concrete names.